client/packages/system/src/Patient/JsonForms/components/BMI.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useEffect, useState } from 'react';
+import {
+  composePaths,
+  ControlProps,
+  rankWith,
+  uiTypeIs,
+} from '@jsonforms/core';
+import { withJsonFormsControlProps } from '@jsonforms/react';
+import {
+  DetailInputWithLabelRow,
+  FORM_LABEL_WIDTH,
+  NumUtils,
+} from '@openmsupply-client/common';
+
+export const bmiTester = rankWith(10, uiTypeIs('BMI'));
+
+const round = (value: number) => Math.round(value * 100) / 100;
+
+const UIComponent = (props: ControlProps) => {
+  const { data, handleChange, label, path } = props;
+  const [bmi, setBmi] = useState<number | null>(null);
+
+  useEffect(() => {
+    if (!data) return;
+
+    const height = NumUtils.parseString(data.height);
+    const weight = NumUtils.parseString(data.weight);
+
+    if (!height || !weight) return;
+
+    setBmi(round(weight / (height * height)));
+  }, [data]);
+
+  useEffect(() => {
+    if (bmi === null) return;
+    handleChange(composePaths(path, 'bodyMassIndex'), bmi);
```

Comment:
I see this is similar to dateOfBirth where `targetField` is hard coded. I guess it works, but a bit worried that need to look inside a component code to figure out which json property it maps to.

Also was wondering about this difference:

https://github.com/openmsupply/open-msupply/blob/b8191643ae8ec02999f6881e684e3154b606d157/server/service/src/sync/program_schemas/patient_ui_schema.json#L105-L112

and 

https://github.com/openmsupply/open-msupply/blob/b8191643ae8ec02999f6881e684e3154b606d157/server/service/src/sync/program_schemas/hiv_care_encounter_ui_schema.json#L29-L36

Where dob doesn't have the `properties` in scope and `BMI` does

Reply:
yeah. it isn't ideal.
the date of birth is given the root scope, and I didn't see that as necessary for bmi. there is an assumption that the weight and height will be siblings of bmi; but I think that is valid. it would be weird to have it update via a different screen, which is how we've broken up the documents.